lib/fastlane/setup/setup_ios.rb

```diff
@@ -22,29 +22,49 @@ def run
 
       show_infos
 
+      fastlane_actions_path = File.join(FastlaneFolder.path, 'actions')
+      is_manual_setup = false
+
       begin
         FastlaneFolder.create_folder! unless Helper.is_test?
         setup_project
         ask_for_apple_id
         detect_if_app_is_available
         print_config_table
-        fastlane_actions_path = File.join(FastlaneFolder.path, 'actions')
         if UI.confirm("Please confirm the above values")
           default_setup(path: fastlane_actions_path)
         else
+          is_manual_setup = true
           manual_setup(path: fastlane_actions_path)
         end
         Helper.log.info 'Successfully finished setting up fastlane'.green
       rescue => ex # this will also be caused by Ctrl + C
-        # Something went wrong with the setup, clear the folder again
-        # and restore previous files
-        Helper.log.fatal 'Error occurred with the setup program! Reverting changes now!'.red
-        restore_previous_state
-        raise ex
+        if is_manual_setup
+          handle_exception(ex)
+        else
+          Helper.log.error 'An error occured during the setup process. Reverting to manual setup!'.yellow
+          try_manual_setup(path: fastlane_actions_path)
+        end
       end
       # rubocop:enable Lint/RescueException
     end
 
+    def handle_exception(exception: nil)
+      # Something went wrong with the setup, clear the folder again
+      # and restore previous files
+      Helper.log.fatal 'Error occurred with the setup program! Reverting changes now!'.red
+      restore_previous_state
+      raise exception
+    end
+
+    def try_manual_setup(path: nil)
+      begin
```

Comment:
What @houndci-bot means here is that in Ruby you can remove the `begin` block when you rescue a method
```ruby
def something
  # do something dangerous
rescue => ex
  puts "yolo"
end
```